maas/resource_maas_raid.go

```diff
@@ -0,0 +1,528 @@
+package maas
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"slices"
+	"strconv"
+	"strings"
+
+	"github.com/canonical/gomaasclient/client"
+	"github.com/canonical/gomaasclient/entity"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceMAASRAID() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Provides a resource to manage MAAS RAIDS, and construct them from block devices and partitions.",
+		CreateContext: resourceRAIDCreate,
+		ReadContext:   resourceRAIDRead,
+		UpdateContext: resourceRAIDUpdate,
+		DeleteContext: resourceRAIDDelete,
+
+		Schema: map[string]*schema.Schema{
+			"block_devices": {
+				Type:         schema.TypeSet,
+				Optional:     true,
+				Elem:         &schema.Schema{Type: schema.TypeString},
+				Description:  "The list of block devices to be included in the RAID.\n*Note*: The boot disk cannot participate in the RAID as a block device, a partition on top of it should be supplied instead.\n*Note*: Block devices with partitions are not valid targets to construct a raid, supply their partitions instead.",
+				AtLeastOneOf: []string{"block_devices", "partitions"},
+			},
+			"fs_type": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The file system type (e.g. `ext4`). If this is not set, the RAID is unformatted.",
+			},
+			"level": {
+				Type:     schema.TypeString,
+				Required: true,
+				// TODO: Re-add RAID-10 once the fix for LP#2109708 is released
```

Comment:
I do not think this is the proper solution. Even when the fix is landed and released, then RAID-10 will not throw the error from MAAS versions >= version-with-the-fix. Let me propose some alternatives:

- Do not include the calculated size in the Terraform fields for any RAID
- Apply the fix at the gomaasclient so that no matter what MAAS version is used, both the Go client and Terraform are operational. Then release the client with the fix and finally consume it from the Terraform provider. Fixes to consider for the client: a) make the size a string or a float as the return type. b) consume it as a string from MAAS HTTP API and then try to return it as int

My 2 cents: I would prefer fixing the client